packages/gatsby-transformer-remark/src/extend-node-type.js

```diff
@@ -165,10 +168,22 @@ module.exports = (
       }
     }
 
-    async function getMarkdownAST(markdownNode) {
+    async function getMarkdownAST(content, markdownNode) {
       if (process.env.NODE_ENV !== `production` || !fileNodes) {
         fileNodes = getNodesByType(`File`)
       }
+
+      // compiler to inject in the remark plugins
+      // so that they can use our parser/generator
+      // with all the options and plugins from the user
+      const compiler = {
+        parseString: string => getMarkdownAST(string, markdownNode),
```

Comment:
Multiple calls of `getMarkdownAST` with the same `markdownNode` may potentially cause problems because of `mutateSource`. Other plugins literally mutate `markdownNode` inside this function. For example:

https://github.com/Rulikkk/gatsby-remark-emoji/blob/8dd4f880b54e0a83ab2dba924fee58550a570a87/index.js#L9-L22

After this change, they will apply mutation to the same `markdownNode` (not `string`) multiple times. There may be idempotent plugins (maybe even the example above) but in general, idempotency is not guaranteed in `mutateSource`.

Reply:
Ah interesting, I didn't consider that